ci-operator/step-registry/upi/conf/vsphere/upi-conf-vsphere-commands.sh

```diff
@@ -0,0 +1,149 @@
+vm_template="${ova_url##*/}"
+
+
+echo "$(date -u --rfc-3339=seconds) - Creating govc.sh file..."
+cat >> "${SHARED_DIR}/govc.sh" << EOF
+export GOVC_URL=vcsa-ci.vmware.devcluster.openshift.com
+export GOVC_USERNAME="${vsphere_user}"
+export GOVC_PASSWORD="${vsphere_password}"
+export GOVC_INSECURE=1
+export GOVC_DATACENTER=dc1
+export GOVC_DATASTORE=vsanDatastore
+EOF
+
+echo "$(date -u --rfc-3339=seconds) - Extend install-config.yaml ..."
+
+# We are not setting
+# machineCIDR: 139.178.94.128/25
+# does this cause an issue?
+cat >> "${install_config}" << EOF
+baseDomain: $base_domain
+controlPlane:
+  name: "master"
+  replicas: 3
+compute:
+- name: "worker"
+  replicas: 0
+platform:
+  vsphere:
+    cluster: devel
+    datacenter: dc1
+    defaultDatastore: vsanDatastore
+    network: "VM Network"
+    password: ${vsphere_password}
+    username: ${vsphere_user}
+    vCenter: vcsa-ci.vmware.devcluster.openshift.com
+    folder: "/dc1/vm/${cluster_name}"
+EOF
+
+echo "$(date -u --rfc-3339=seconds) - Create terraform.tfvars ..."
+cat > "${SHARED_DIR}/terraform.tfvars" <<-EOF
+machine_cidr = "139.178.94.128/25"
+vm_template = "${vm_template}"
+vsphere_cluster = "devel"
+vsphere_datacenter = "dc1"
+vsphere_datastore = "vsanDatastore"
+vsphere_server = "vcsa-ci.vmware.devcluster.openshift.com"
+ipam = "139.178.89.254"
+cluster_id = "${cluster_name}"
+base_domain = "${base_domain}"
+cluster_domain = "${cluster_domain}"
+ssh_public_key_path = "${ssh_pub_key_path}"
+compute_memory = "16384"
+compute_num_cpus = "4"
+EOF
+
+dir=/tmp/installer
+mkdir "${dir}/"
+pushd ${dir}
+cp -t "${dir}" \
+    "${SHARED_DIR}/install-config.yaml"
+
+### Create manifests
+echo "Creating manifests..."
+openshift-install --dir="${dir}" create manifests &
+
+set +e
+wait "$!"
+ret="$?"
+set -e
+
+if [ $ret -ne 0 ]; then
+  cp "${dir}/.openshift_install.log" "${ARTIFACT_DIR}/.openshift_install.log"
+  exit "$ret"
+fi
+
+### Remove control plane machines
+echo "Removing control plane machines..."
+rm -f openshift/99_openshift-cluster-api_master-machines-*.yaml
+
+### Remove compute machinesets (optional)
+echo "Removing compute machinesets..."
+rm -f openshift/99_openshift-cluster-api_worker-machineset-*.yaml
+
+### Make control-plane nodes unschedulable
+echo "Making control-plane nodes unschedulable..."
+sed -i "s;mastersSchedulable: true;mastersSchedulable: false;g" manifests/cluster-scheduler-02-config.yml
+
+### Create Ignition configs
+echo "Creating Ignition configs..."
+openshift-install --dir="${dir}" create ignition-configs &
+
+set +e
+wait "$!"
+ret="$?"
+set -e
+
+cp "${dir}/.openshift_install.log" "${ARTIFACT_DIR}/.openshift_install.log"
+
+if [ $ret -ne 0 ]; then
+  exit "$ret"
+fi
+
+cp -t "${SHARED_DIR}" \
+    "${dir}/auth/kubeadmin-password" \
+    "${dir}/auth/kubeconfig" \
+    "${dir}/metadata.json" \
+    ${dir}/*.ign
+
+# Removed tar of openshift state. Not enough room in SHARED_DIR with terraform state
```

Comment:
did we end up creating something for test platform team to help fix this. the 1mb restriction is not helping UPI workflows.

Reply:
Its a kubernetes limitation for a secret I am not sure how we would be working around that. I might be missing something but for vSphere UPI not sure why we need openshift state.

cc: @wking @stevekuznetsov

Reply:
Tucking the metadata into the cluster itself is one option, but maybe install failed.  In that case, I think the option are:

* Tucking the data into a separate storage engine like S3 (I think Ahbinav also floated matchbox?).
* Reducing the amount of data that you need to share.  Can we teardown with some subset of the Terraform state?

And it's not clear to me why we'd want to share the installer's state file between steps, even if we didn't have to make room for a large Terraform state dump.

Also in this space, CI jobs can fail if a CI node goes down, which happens sometimes.  We'll want a stateless way to reap the resources that sort of hard-termination leaks, and it's possible we can recycle some of that approach for the official teardown step to reduce the amount of state it needs passed in.